packages/ncids-js/src/components/nci-header/utils/megamenu.ts

```diff
@@ -0,0 +1,557 @@
+			.catch((error) => {
+				console.error(`Fetch Error: ${id} ${error}`);
+				return [];
+			});
+		return await response;
+	}
+
+	/**
+	 * Returns array of LI elements in the primary navigation
+	 *
+	 * @return {NodeListOf<HTMLElement>}
+	 * @private
+	 */
+	private queryListSections(): HTMLElement[] {
+		const selector = this.options.listElementClass;
+		return Array.from(this.element.querySelectorAll(<string>selector));
+	}
+
+	/**
+	 * Returns the HTMLElement that contains the MegaMenu
+	 *
+	 * @return {<HTMLElement>} MegaMenu DOM Element
+	 * @private
+	 */
+	private queryMegaMenu(button: Element): HTMLElement {
+		const selector = '.nci-header-megamenu';
+		const parent = <HTMLElement>button.parentElement;
+		return <HTMLElement>parent.querySelector(selector);
+	}
+
+	/**
+	 * Creates the primary navigation buttons from the link
+	 * element passed into it and adds aria tags
+	 *
+	 * @return {<HTMLElement>}  Button element with proper tags
+	 *
+	 * @param {Element} link anchor tag to be converted into button.
+	 * @private
+	 */
+	private activateButton(link: Element): Element {
+		const id = link.getAttribute('href') || '';
+		const button = this.createDom(
+			'button',
+			['nci-header-nav__primary-button'],
+			[{ 'aria-expanded': false }, { 'aria-controls': id }]
+		);
+		button.innerHTML = link.innerHTML;
+		link.replaceWith(button);
+		return link;
+	}
+
+	/**
+	 * Change the primary buttons back to links
+	 *
+	 * @param {Element} button anchor tag to be converted into button.
+	 * @return {<HTMLElement>}  Button element with proper tags
+	 * @private
+	 */
+	private deactivateButton(button: Element): Element {
+		const link = this.createDom('a', ['nci-header-nav__primary-link']);
+		link.innerHTML = button.innerHTML;
+		button.replaceWith(link);
+		return link;
+	}
+
+	/**
+	 * Finds newly updated buttons and adds event listeners.
+	 *
+	 * @private
+	 */
+	private findToggleButtons(): void {
+		const sections = this.queryListSections();
+		sections.forEach((section) => {
+			const button = section.firstElementChild;
+			this.toggleButtons.push(<HTMLElement>button);
+		});
+		this.addToggleEventListeners();
+	}
+
+	/**
+	 * Sets up event listeners for every possible button per collapsible.
+	 *
+	 * @private
+	 */
+	private addToggleEventListeners(): void {
+		[...this.toggleButtons].forEach((button) => {
+			button.addEventListener('click', this.eventListener);
+		});
+	}
+
+	/**
+	 * Removes event listeners for every possible button per collapsible.
+	 *
+	 * @private
+	 */
+	private removeToggleEventListeners(): void {
+		[...this.toggleButtons].forEach((button) => {
+			button.removeEventListener('click', this.eventListener);
+		});
+	}
+
+	/**
+	 * Click handler for the toggle buttons.
+	 *
+	 * @param {Event} event Collapsible section toggling event.
+	 *
+	 * @private
+	 */
+	private handleClick(event: Event): void {
+		event.preventDefault();
+		event.stopPropagation();
+		if (event.target instanceof HTMLElement) {
```

Comment:
This check is also throwing an error. I had a similar problem in site alert and footer where I got rid of the need to pass button through the method, and instead made button a property, so I didn't need to create more tests for both cases. Funnily enough, it looks like you may be able to easily fix this by changing HTMLElement to HTMLButtonElement since you built tests for non-js header already.

```ts
		if (event.target instanceof HTMLButtonElement) {
```